### PR TITLE
W-11412423: Make Extension model declare components min mule version

### PR DESCRIPTION
**Pull request by lbarrios**

### CODE REVIEW

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/SinceMuleVersionTestCase.java [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    JavaExtensionModelParserFactory parserFactory =
+        createExtensionParserFactoryForSinceMuleVersionModelProperty(loadingContext, property);
+
+    DefaultExtensionModelLoaderDelegate delegate = new DefaultExtensionModelLoaderDelegate(getProductVersion());
+    ExtensionDeclarer extensionDeclarer = delegate.declare(parserFactory, loadingContext);
+    return extensionDeclarer.getDeclaration();
+  }
+
+  private void assertOptionalsAreEqual(Optional<?> p1, Optional<?> p2) {
+    assertThat(p1.isPresent()).isEqualTo(p2.isPresent());
```
Comment (ndinu) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/SinceMuleVersionTestCase.java:147]: Use hamcrest for assertions.

Comment (lbarrios) [reply modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/SinceMuleVersionTestCase.java]: Done

**Review by ndinu — APPROVED** [review]

**Comment by lbarrios** [comment]
--validate

**Comment by lbarrios** [comment]
--validate

**Review by ndinu — APPROVED** [review]

**Pull request merged**